cpp/include/raft/core/detail/buffer_utils/copy_cpu.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <algorithm>
+#include <cstring>
+#include <raft/core/device_support.hpp>
+#include <raft/core/device_type.hpp>
+#include <raft/core/execution_stream.hpp>
+
+namespace raft {
+namespace detail {
+
+template <device_type dst_type, device_type src_type, typename T>
+std::enable_if_t<std::conjunction_v<std::bool_constant<dst_type == device_type::cpu>,
+                                    std::bool_constant<src_type == device_type::cpu>>,
+                 void>
+copy(T* dst, T const* src, uint32_t size, execution_stream stream)
+{
+  std::copy(src, src + size, dst);
+}
+
+template <device_type dst_type, device_type src_type, typename T>
+std::enable_if_t<
+  std::conjunction_v<std::disjunction<std::bool_constant<dst_type != device_type::cpu>,
+                                      std::bool_constant<src_type != device_type::cpu>>,
+                     std::bool_constant<!CUDA_ENABLED>>,
```

Comment:
We have a boolean defined [here](https://github.com/rapidsai/raft/pull/1402/files#diff-2198de5ee72752622ef5dc656109c6263e114bd3a99eefcde4b946d6f5a9065aR22), whereas `RAFT_DISABLE_CUDA` is a directive. So I'm not sure if it can be used to evaluate boolean expressions.